src/test/java/com/baloise/confluence/digitalsignature/DigitalSignatureMacroTest.kt

```diff
@@ -0,0 +1,120 @@
+        override fun init() {}
+
+        override fun setValue(p0: BandanaContext?, p1: String?, p2: Any?) {
+            content[p1!!] = p2!!
+        }
+
+        override fun getValue(p0: BandanaContext?, p1: String?): Any? {
+            return content[p1!!]
+        }
+
+        override fun getValue(p0: BandanaContext?, p1: String?, p2: Boolean): Any? {
+            return content[p1!!]
+        }
+
+        override fun getKeys(p0: BandanaContext?): MutableIterable<String> {
+            return content.keys
+        }
+
+        override fun removeValue(p0: BandanaContext?, p1: String?) {
+            content.remove(p1)
+        }
+
+    }
+
+    @Test
+    fun mailtoLong(): Unit {
+        val macro = DigitalSignatureMacro()
+        val profiles: MutableList<UserProfile> = ArrayList()
+        val profile = buildUserProfile("Heinz Meier", "heinz.meier@meier.com")
+        for (i in 0..19) {
+            profiles.add(profile)
+        }
+
+        val mailto = macro.getMailto(profiles, "Subject", true, null)
+
+        Assertions.assertEquals(
+            "mailto:heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com,heinz.meier@meier.com?Subject=Subject",
+            mailto
+        )
+    }
+
+    @Test
+    fun mailtoVeryLong(): Unit {
+        val macro =
+            DigitalSignatureMacro()
+        val profiles: MutableList<UserProfile> = ArrayList()
+        val profile = buildUserProfile("Heinz Meier", "heinz.meier@meier.com")
+        for (i in 0..199) {
+            profiles.add(profile)
+        }
+
+        val mailto = macro.getMailto(profiles, "Subject", true, signature, "nirvana")
+
+        Assertions.assertEquals(
+            "nirvana/rest/signature/1.0/emails?key=signature.3224a4d6bba68cd0ece9b64252f8bf5677e24cf6b7c5f543e3176d419d34d517&signed=true",
+            mailto
+        )
+    }
+
+    @Test
+    fun mailtoShort(): Unit {
+        val macro = DigitalSignatureMacro()
+        val profiles: MutableList<UserProfile> = ArrayList()
+        val profile = buildUserProfile("Heinz Meier", "heinz.meier@meier.com")
+        profiles.add(profile)
+
+        val mailto = macro.getMailto(profiles, "Subject", true, null)
+
+        Assertions.assertEquals(
+            "mailto:Heinz Meier<heinz.meier@meier.com>?Subject=Subject",
+            mailto
+        )
+    }
+
+    fun buildUserProfile(userName: String, userEmail: String)= object : UserProfile {
+            override fun getFullName(): String = userName
+            override fun getEmail(): String = userEmail
+
+            override fun getUserKey(): UserKey {
+                TODO("Not yet implemented")
```

Comment:
TODO will show up in tools like IntelliJ TODO list window.
Better to use `throw UnsupportedOperationException("Not yet implemented")`